File: Dataset/Leetcode/train/125/161.py
 class Solution:
    def XXX(self, s: str) -> bool:
        i, j = 0, len(s)-1
        while i < j:
            if not s[i].isalnum():
                i += 1
                continue
            if not s[j].isalnum():
                j -= 1
                continue

            if s[i].lower() == s[j].lower():
                i += 1
                j -= 1
                continue
            return False
        return True

File: Dataset/Leetcode/train/3/161.java
class Solution {
    public int XXX(String s) {
        int len = 1;
        int index = 0;
        if(s.equals("")) return 0;
        for (int i = 0; i < s.length(); i++) {
            String sc = String.valueOf(s.charAt(i));
            int indexOf = s.substring(index, i).indexOf(sc)+index;
            if (indexOf>=0) index = indexOf+1;
            len = Math.max(s.substring(index, i+1).length(), len);
        }
        return len;
    }
}

